include/boost/math/special_functions/fourier_transform_daubechies_scaling.hpp

```diff
@@ -0,0 +1,247 @@
+                                -0.00960430880128020906860390254555211461150702751378997239464015046967050703218076318595987803)};
+  }
+}
+
+} // namespace detail
+
+/*
+ * Given ω∈ℝ, computes a numerical approximation to 𝓕[𝜙](ω),
+ * where 𝜙 is the Daubechies scaling function.
+ * Fast and accurate evaluation of these function seems to me to be a rather involved research project,
+ * which I have not endeavored to complete.
+ * In particular, recovering ~1ULP evaluation is not possible using the techniques
+ * employed here-you should use this with the understanding it is good enough for almost
+ * all uses with empirical data, but probably doesn't recover enough accuracy
+ * for pure mathematical uses (other than graphing-in which case it's fine).
+ * The implementation uses an infinite product of trigonometric polynomials.
+ * See Daubechies, 10 Lectures on Wavelets, equation 5.1.17, 5.1.18.
+ * It uses the factorization of m₀ shown in Corollary 5.5.4 and equation 5.5.5.
+ * See more discusion near equation 6.1.1,
+ * as well as efficiency gains from equation 7.1.4.
+ */
+template <class Real, unsigned p> std::complex<Real> fourier_transform_daubechies_scaling(Real omega) {
+  // This arg promotion is kinda sad, but IMO the accuracy is not good enough in
+  // float precision using this method. Requesting a better algorithm!
+  if constexpr (std::is_same_v<Real, float>) {
+    return static_cast<std::complex<float>>(fourier_transform_daubechies_scaling<double, p>(static_cast<double>(omega)));
+  }
+  using boost::math::constants::one_div_root_two_pi;
+  using std::abs;
+  using std::exp;
+  using std::norm;
+  using std::pow;
+  using std::sqrt;
+  using std::cbrt;
+  // Equation 7.1.4 of 10 Lectures on Wavelets is singular at ω=0:
+  if (omega == 0) {
+     return std::complex<Real>(one_div_root_two_pi<Real>(), 0);
+  }
+  // For whatever reason, this starts returning NaNs rather than zero for |ω|≫1.
+  // But we know that this function decays rather quickly with |ω|,
+  // and hence it is "numerically zero", even if in actuality the function does not have compact support.
+  // Now, should we probably do a fairly involved, exhaustive calculation to see where exactly we should set this threshold
+  // and store them in a table? .... yes.
+  if (abs(omega) >= sqrt(std::numeric_limits<Real>::max())) {
+       return std::complex<Real>(0, 0);
+  }
+  auto const constexpr lxi = detail::ft_daubechies_scaling_polynomial_coefficients<Real, p>();
+  auto xi = -omega / 2;
+  std::complex<Real> phi{one_div_root_two_pi<Real>(), 0};
+  do {
+    std::complex<Real> arg{0, xi};
+    auto z = exp(arg);
+    phi *= boost::math::tools::evaluate_polynomial_estrin(lxi, z);
+    xi /= 2;
+  } while (abs(xi) > std::numeric_limits<Real>::epsilon());
+  std::complex<Real> arg{0, omega};
+  // There is no std::expm1 for complex numbers.
+  // We may therefore be leaving accuracy gains on the table for small |ω|:
```

Comment:
Is it worth adding the complex case to Boost.Math::expm1? This Julia PR doing the same thing makes it seem not overly difficult https://github.com/JuliaLang/julia/pull/6904/files.